syntax = "proto3";

package distance_tracking;

service DistanceServer {

    rpc registerClient (ClientInfo) returns (ServerInfo) {
    }

    rpc getDistances (ClientInfo) returns (stream DistanceValue) {
    }

}

message ClientInfo {
    string info = 1;
}

message ServerInfo {
    string info = 1;
}

message DistanceValue {
    int32 id = 1;
    int64 ts = 2;
    int64 elapsed = 3;
    int32 distance = 4;
}


